include/seqan3/contrib/charconv/charconv_pre.hpp

```diff
@@ -0,0 +1,57 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <cassert>
+#include <system_error>
+#include <limits>
+
+#include <seqan3/core/platform.hpp>
+
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_BEGIN_NAMESPACE_STD namespace seqan3::contrib::charconv {
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_END_NAMESPACE_STD }
+// #define SEQAN3_CONTRIB_CHARCONV_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER 1 // needs to be set if compiling with MSVC
+// #define SEQAN3_CONTRIB_CHARCONV_LIBCPP_COMPILER_MSVC // needs to be set if compiling with MSVC
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_STD_VER 17
+// #define SEQAN3_CONTRIB_CHARCONV_LIBCPP_PUSH_MACROS
+// #define SEQAN3_CONTRIB_CHARCONV_LIBCPP_POP_MACROS
+// #define SEQAN3_CONTRIB_CHARCONV_LIBCPP_AVAILABILITY_TO_CHARS
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_FUNC_VIS
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_ENUM_VIS
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_TYPE_VIS
+// #define SEQAN3_CONTRIB_CHARCONV_LIBCPP_CONSTEXPR constexpr
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_HIDDEN
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_INLINE_VISIBILITY
+#define SEQAN3_CONTRIB_CHARCONV_LIBCPP_ASSERT(condition, message) assert(condition)
```

Comment:
Can you shortly explain the neccessity of these macros? (I am not doubting them, just not understanding them)

Reply:
These are all prefixed MACROS that the original implementation https://github.com/llvm/llvm-project/blob/d27489645bb8829a952f79e3b211202844238391/libcxx/include/charconv uses.

Our previous approach of renaming and seqan3-ifing external things was wrong, because an update is REALLY hard to do. The new approach is to keep the original sources as close as they are and only do changes that are deemed necessary.